mlrun/api/db/sqldb/db.py

```diff
@@ -1222,6 +1227,7 @@ def tag_artifacts(self, session, artifacts, project: str, name: str):
             )
             tag = query.one_or_none()
             if not tag:
+                validate_tag_name(tag_name=name, field_name="artifact.metadata.tag")
                 tag = artifact.Tag(project=project, name=name)
```

Comment:
Please add a comment why the validation is being done only if tag didn't exists ( backwards compatibility etc.. )